internal/util/search/search.go

```diff
@@ -231,3 +258,36 @@ func (sr *SearchReplace) settersValues() (map[string]string, error) {
 func clean(input string) string {
 	return strings.Trim(strings.TrimSpace(input), `"`)
 }
+
+// Validate validates the yaml content against kubernetes schema used by kubeval
+func ValidateNode(object *yaml.RNode) ([]kubeval.ValidationResult, error) {
+	objStr, err := object.String()
+	if err != nil {
+		return nil, err
+	}
+	config := kubeval.NewDefaultConfig()
+	return kubeval.Validate([]byte(objStr), config)
+}
+
+// deepCopy deep copies an RNode object
+func deepCopy(object *yaml.RNode) (*yaml.RNode, error) {
+	objStr, err := object.String()
+	if err != nil {
+		return nil, err
+	}
+	return yaml.Parse(objStr)
+}
+
+// filterAll runs the yaml.Filter against all inputs
+func filterAll(filter yaml.Filter) kio.Filter {
```

Comment:
Might be worth explaining why we need this and how it is different from the `kio.FilterAll` function: https://github.com/kubernetes-sigs/kustomize/blob/052a6b4e967b8cba7f9c12623c9dd53fb16e2f7a/kyaml/kio/kio.go#L139